src/Editor.js

```diff
@@ -1332,98 +1340,132 @@ const Editor = () => {
                 config[
                   doc.layers.find(({ id }) => id === [...view.selection][0])
                     .component
-                ].options?.map(({ key, input, label }) => {
+                ].options?.map(({ key, input, label }, index) => {
+                  const layer = doc.layers.find(
+                    ({ id }) => id === [...view.selection][0]
+                  );
+                  const error = config[layer.component].validate(
+                    layer.options,
+                    key
```

Comment:
```suggestion
```

Reply:
I made the change and a few other changes that I must have forgotten to include